src/Minsk/CodeAnalysis/Lowering/Lowerer.cs

```diff
@@ -263,12 +222,145 @@ protected override BoundStatement RewriteConditionalGotoStatement(BoundCondition
                 var condition = (bool)node.Condition.ConstantValue.Value;
                 condition = node.JumpIfTrue ? condition : !condition;
                 if (condition)
-                    return RewriteStatement(new BoundGotoStatement(node.Label));
+                    return RewriteStatement(Goto(node.Label));
                 else
-                    return RewriteStatement(new BoundNopStatement());
+                    return RewriteStatement(Nop());
             }
 
             return base.RewriteConditionalGotoStatement(node);
         }
+
+        #region BoundNodeFactory
+
+        private BoundNopStatement Nop()
```

Comment:
Can we extract this to a separate type `BoundNodeFactory`? We can import these methods in `Lowerer.cs` via `using static Minsk.CodeAnalsysis.Binding.BoundNodeFactory;`